### PR TITLE
Drop Python 2 intersphinx URL

### PR DESCRIPTION
**Pull request by mthuurne**
## Contributor Checklist:

* [x] The associated ticket in Trac is here: https://twistedmatrix.com/trac/ticket/10042
* [x] I ran `tox -e lint` to format my patch to meet the [Twisted Coding Standard](https://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html)
* [x] I ran additional checks listed at [Getting Your Patch Accepted](https://twistedmatrix.com/trac/wiki/TwistedDevelopment#GettingYourPatchAccepted)
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: [News files](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles))
* [x] I have updated the automated tests.
* [x] I have submitted the associated Trac ticket for review by adding the word `review` to the keywords field in Trac, and putting a link to this PR in the comment; it shows up in https://twisted.reviews/ now.

### CODE REVIEW

File: src/twisted/logger/_io.py [hunk 1]
```diff
@@ -22,11 +24,19 @@ class LoggingFile:
     Note that because event formats are L{str}, C{bytes} received via C{write()}
     are converted to C{str}, which is the opposite of what C{file} does.
 
-    @ivar softspace: File-like L{'softspace' attribute <file.softspace>}; zero
-        or one.
+    @ivar softspace: Attribute to make this class more file-like under Python 2;
+        value is zero or one.  Do not use.
```
Comment (adiroiban) [inline src/twisted/logger/_io.py:28]: Instead of "Do not use" maybe just say "Deprecated." :)

**Review by adiroiban — COMMENTED** [review]
This looks good. Thanks for the cleanup 

Will try to create a test for the deprecation part.

**Review by adiroiban — APPROVED** [review]
Assuming that the tests are green (with the exception of mypy3) I am happy and I think that this can be merged.

Thanks!

**Comment by mthuurne** [comment]
It seems the reruns necessary to get the build to pass on macOS broke the RTD task: it is marked as failed even though there are no errors in its log.

**Pull request merged**